content/post/codesent.md

```diff
@@ -0,0 +1,395 @@
+```
+
+We emulate the neutral sentiment class as the middle between positiveness and
+negativeness and exclude them from further analysis. The results are shown on the table.
+
+| Project               | Positive | Negative | Effective |
+|-----------------------|----------|----------|-----------|
+| [pygame/pygame](https://github.com/pygame/pygame)         | 93.4     | 98.2     | -4.8      |
+| [pallets/flask](https://github.com/pallets/flask)         | 23.6     | 31.9     | -8.3      |
+| [django/django](https://github.com/django/django)         | 462.1    | 536.1    | -74.0     |
+| [golang/go](https://github.com/golang/go)             | 402.1    | 507.3    | -105.2    |
+| [kubernetes/kubernetes](https://github.com/kubernetes/kubernetes) | 186.1    | 91.6     | 94.5      |
+| [keras-team/keras](https://github.com/keras-team/keras)      | 104.8    | 71.1     | 33.7      |
+
+The columns represent the sum of all the positive and negative sentiments together with the final
+sentiment value. Some of the projects appeared to be negative and some positive. We'll see why
+later, and now let's look at the promised sentiment plot through time.
+
+{{% caption src="/post/codesent/golang-sentiment-black.png" %}}
+golang/go sentiment through time.
+{{% /caption %}}
+
+Can we conclude anything useful from this? Unfortunately, no, there are no clear trends or
+periods. It is too many different comments. Maybe we will have more luck with monorepos, that is
+those with a single main contributor. For now, it is more fun to inspect the most negative and positive
+examples of comments on Golang source code. Negative:
+
+```go
+// It is low-level, old, and unused by Go's current HTTP stack
+
+// not clear why this shouldn't work
+
+// overflow causes something awful
+
+// quiet expected TLS handshake error remote error: bad certificate
+```
+Source: [1](https://github.com/golang/go/blob/master/src/net/http/httputil/persist.go#L33),
+[2](https://github.com/golang/go/blob/32aa0d9198a1b147eb2c756c0d5628b3399f9898/test/bugs/bug046.go#L11),
+[3](https://github.com/golang/go/blob/master/src/runtime/vlrt.go#L95),
+[4](https://github.com/golang/go/blob/master/src/os/exec/exec_test.go#L568).
+
+Positive:
+
+```go
+// Special case, useful in debugging
+
+// Implementation: Parallel summing of adjacent bits. See Hackers Delight, Chap. 5: Counting Bits.
+
+// TODO(austin): This could be a really awesome string method
+
+// TODO(bradfitz): this might be 0, once escape analysis is better
+```
+
+Source: [1](https://github.com/golang/go/blob/master/src/bytes/buffer.go#L63),
+[2](https://github.com/golang/go/blob/master/src/math/bits/bits.go#L135),
+[3](https://github.com/golang/go/blob/46e392e01c630dee41a67e01223b538aae9dc9b5/usr/austin/ogle/vars.go#L42),
+[4](https://github.com/golang/go/blob/84ef97b59c89b7d9fdc04a1a8a438cd3257bf521/src/pkg/strconv/strconv_test.go#L21).
+
+The last one is actually written by Brad who talked a few minutes ago hehe. In this case the model
```

Comment:
s/who talked a few minutes ago hehe./who gave the Gophercon talk right before me hehe./

Reply:
It was Brad then Brian then me...